drivers/bbram/bbram_npcx.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_bbram
+
+#include <drivers/bbram.h>
+#include <errono.h>
+#include <string.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
+
+/** Device config */
+struct bbram_npcx_config {
+	/** BBRAM base address */
+	uintptr_t base_addr;
+	/** BBRAM size (Unit:bytes) */
+	int size;
+	/** Status register base address */
+	uintptr_t status_reg_addr;
+};
+
+#define NPCX_STATUS_IBBR BIT(7)
+#define NPCX_STATUS_VSBY BIT(1)
+#define NPCX_STATUS_VCC1 BIT(0)
+
+#define DRV_CONFIG(dev) ((const struct bbram_npcx_config *)(dev)->config)
+#define DRV_STATUS(dev)                                                        \
+	(*((volatile uint8_t *)DRV_CONFIG(dev)->status_reg_addr))
+
+static int get_bit_and_reset(const struct device *dev, int mask)
+{
+	int result = DRV_STATUS(dev) & mask;
+
+	/* Clear the bit(s) */
+	DRV_STATUS(dev) |= mask;
```

Comment:
Oops, sorry for the typo. It should be `DRV_STATUS(dev) = mask;` The original one may clear the other bits.